### PR TITLE
Atomic pairing for SET NX and EXPIRE on resource locks

### PR DESCRIPTION
**Pull request by cjus**
Both commands (SET NX and EXPIRE) are executed in a single atomic transaction. There's no possibility of another process interfering between the two operations

### CODE REVIEW

**Review by callicles — APPROVED** [review]

File: apps/framework-cli/src/infrastructure/redis/redis_client.rs [hunk 1]
```diff
@@ -164,32 +164,23 @@ impl RedisClient {
...
     pub async fn attempt_lock(&self, name: &str) -> Result<bool> {
+        info!("<RedisClient> Attempting to lock {}", name);
         if let Some(lock) = self.locks.get(name) {
-            let result: bool = self
-                .connection
-                .lock()
-                .await
+            let result = redis::pipe()
+                .atomic()
                 .set_nx(&lock.key, &self.instance_id)
```
Comment (phiSgr) [inline apps/framework-cli/src/infrastructure/redis/redis_client.rs:179]: what about using `SetOptions` with `ExistenceCheck::NX` and `SetExpiry::EX(60)`?

File: apps/framework-cli/src/cli/routines.rs [hunk 2]
```diff
@@ -364,7 +364,11 @@ fn start_leadership_lock_task(redis_client: Arc<Mutex<RedisClient>>, project: Ar
         loop {
             interval.tick().await;
             if let Err(e) = manage_leadership_lock(&redis_client, &project).await {
-                error!("Error managing leadership lock: {}", e);
+                if !e.to_string().contains("Failed to attempt lock") {
```
Comment (DatGuyJonathan) [inline apps/framework-cli/src/cli/routines.rs:367]: We might want to use a const for this so whoever throws this error uses the same message. If one side changes, we'll know about the other side.

File: apps/framework-cli/src/infrastructure/redis/redis_client.rs (hunk 1 above)
Comment (cjus) [reply apps/framework-cli/src/infrastructure/redis/redis_client.rs:179]: didn't try or test that approach. We can certainly do that in a new test PR

File: apps/framework-cli/src/cli/routines.rs (hunk 2 above)
Comment (cjus) [reply apps/framework-cli/src/cli/routines.rs:367]: Good call, we can update in a new PR.

**Pull request merged**